### PR TITLE
Toggle suspend resume

### PR DESCRIPTION
**Pull request by ahussein3**
- Closes #[3245](https://github.com/weaveworks/weave-gitops-enterprise/issues/3245)
- Adding Suspend/Resume to all resources table card view
- That's How the UI should look like
image

### CODE REVIEW

File: plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    'namespaces',
+    namespace,
+    gvk.plural,
+    name,
+  ].join('/');
+
+  return basePath;
+};
+
+export function toggleSuspendsuspendRequest(
```
Comment (AlinaGoaga) [inline plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts:26]: Suspend seems to be repeated in the name 🤔

File: plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      gvk,
+      suspend,
+    );
+
+    alertApi.post({
+      message: `${key} request successful`,
+      severity: 'success',
+      display: 'transient',
+    });
+  } catch (e: any) {
```
Comment (AlinaGoaga) [inline plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts:127]: Can we add the Error type here?

**Review by AlinaGoaga — APPROVED** [review]
@ahussein3 This looks good to me 🥳 It would be good to be able to test the resume also - atm all resources come back with `Suspend error: Cannot read properties of undefined (reading 'annotations')` when I try to suspend them

File: plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts (hunk 2 above)
Comment (ahussein3) [reply plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts:127]: typescript whine about it. It must be any or unknown.
https://stackoverflow.com/questions/69021040/why-catch-clause-variable-type-annotation-must-be-any

File: plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      // sync the source
+      await requestToggleSuspendResource(
+        kubernetesApi,
+        resource.sourceRef.name!,
+        resource.sourceRef.namespace || resource.namespace,
+        resource.clusterName,
+        sourceGVK,
+        suspend,
+      );
+    }
```
Comment (foot) [inline plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts:110]: I don't think we want to resume / suspend the source as well as the ks/hr in this case.

Other ks/hr could be relying on the same source and they would stop updating too which might be susprising.

E.g. The `flux` cli only provides the option to update the source for `reconcile` (sync)

```
❯ flux resume kustomization --help | grep with
❯ flux suspend kustomization --help | grep with
❯ flux reconcile kustomization --help | grep with
  flux reconcile kustomization podinfo --with-source
      --with-source   reconcile Kustomization source
```

Comment (bigkevmcd) [reply plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts:110]: Yes, we should only suspend the viewed resource.

File: plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+    // sync the helm release
```
Comment (foot) [inline plugins/backstage-plugin-flux/src/hooks/useToggleSuspendResource.ts:112]:
```suggestion
```

**Review by bigkevmcd — APPROVED** [review]
Thanks!

Looks good

**Pull request merged**